tests/foreman/cli/test_import.py

```diff
@@ -539,11 +547,198 @@ def test_import_users_recovery(self, test_data):
+        for imp_org, transition_record in product(imp_orgs, transition_data):
+            if transition_record['sat5'] == imp_org['organization_id']:
+                imp_org.update({'sat6': transition_record['sat6']})
+        for imp_org in imp_orgs:
+            self.assertEqual(
+                HostCollection.list(
+                    {'organization-id': imp_org['sat6']}
+                ).return_code, 0
+            )
+
+    @data(*import_org_data())
+    def test_reimport_host_collections_default_negative(self, test_data):
+        """@test: Import all System Groups from the default data set
+        (predefined source) as the Host Collections and try to import
+        them again.
+
+        @feature: Repetitive Import Host-Collections
+
+        @assert: 3 Host Collections created
+
+        """
+        files = dict(self.default_dataset[1])
+        files['users'] = update_csv_values(
+            files['users'],
+            test_data,
+            self.default_dataset[0]
+        )
+        files['system-groups'] = update_csv_values(
+            files['system-groups'],
+            test_data,
+            self.default_dataset[0]
+        )
+        # import the prerequisities
+        for result in (
+            Import.organization({'csv-file': files['users']}),
+            Import.host_collection({'csv-file': files['system-groups']}),
+        ):
+            self.assertEqual(result.return_code, 0)
+        transition_data = csv_to_dataset(
+            ssh.command(
+                u'ls ${HOME}/.transition_data/organization*'
+            ).stdout[:-1]
+        )
+        hcollections_before = [
+            HostCollection.list({'organization-id': tr['sat6']}).stdout
+            for tr in transition_data
+        ]
+        self.assertEqual(
+            Import.host_collection(
+                {'csv-file': files['system-groups']}
+            ).return_code, 0
+        )
+        hcollections_after = [
+            HostCollection.list({'organization-id': tr['sat6']}).stdout
+            for tr in transition_data
+        ]
+        self.assertEqual(hcollections_before, hcollections_after)
+
+    @data(*import_org_data())
+    def test_import_host_collections_recovery(self, test_data):
+        """@test: Try to Import Collections with the same name to invoke
+        usage of a recovery strategy (rename, map, none)
+
+        @feature: Import HostCollection Recover
+
+        @assert: 2nd Import will rename the new collections, 2nd import will
+        map them and the 3rd one will result in No Action Taken
+
+        """
+        # prepare the data
+        files = dict(self.default_dataset[1])
+        files['users'] = update_csv_values(
+            files['users'],
+            test_data,
+            self.default_dataset[0]
+        )
+        files['system-groups'] = update_csv_values(
+            files['system-groups'],
+            test_data,
+            self.default_dataset[0]
+        )
```

Comment:
This can be made more concise with a for loop:

```python
        for file_ in ('users', 'system-groups'):
            files[file_] = update_csv_values(
                files[file_],
                test_data,
                self.default_dataset[0]
            )
```